src/containers/Tenant/Diagnostics/Partitions/PartitionsControls/PartitionsControls.tsx

```diff
@@ -0,0 +1,175 @@
+        () =>
+            consumers
+                ? consumers.map((consumer) => ({
+                      value: consumer,
+                      content: consumer,
+                  }))
+                : undefined,
+        [consumers],
+    );
+
+    const columnsToSelect = useMemo(() => {
+        return initialColumnsIds.map((id) => {
+            return {
+                title: PARTITIONS_COLUMNS_TITILES[id as ValueOf<typeof PARTITIONS_COLUMNS_IDS>],
+                selected: Boolean(!hiddenColumns.includes(id)),
+                id: id,
+                required: id === PARTITIONS_COLUMNS_IDS.PARTITION_ID,
+            };
+        });
+    }, [initialColumnsIds, hiddenColumns]);
+
+    const handleConsumerSelectChange = (value: string[]) => {
+        // As we have selector with multiple options, the first value corresponds to previous value
+        // The second value is currently chosen consumer
+        handleSelectedConsumerChange(value[1]);
+        setConsumerSelectOpen(false);
+    };
+
+    const handlePartitionIdSearchChange = (value: string) => {
+        setPartitionIdSearchValue(value);
+    };
+
+    const handleGeneralSearchChange = (value: string) => {
+        setGeneralSearchValue(value);
+    };
+
+    const hadleTableColumnsSetupChange = (value: TableColumnSetupItem[]) => {
+        const result = [...hiddenColumns];
+
+        // Process current set of columns
+        // This way we do not remove from hidden these columns, that are not displayed currently
+        // The reasons: set of columns differs for partitions with and without consumers
+        value.forEach((el) => {
+            if (!el.selected && !hiddenColumns.includes(el.id)) {
+                result.push(el.id);
+            } else if (el.selected && hiddenColumns.includes(el.id)) {
+                result.splice(hiddenColumns.indexOf(el.id));
+            }
+        });
+
+        handleHiddenColumnsChange(result);
+    };
+
+    return (
+        <div className={b('controls')}>
+            <Select
+                className={b('consumer-select')}
+                placeholder={i18n('controls.consumerSelector.placeholder')}
+                label={i18n('controls.consumerSelector')}
+                options={consumersToSelect}
+                value={[selectedConsumer || '']}
+                onUpdate={handleConsumerSelectChange}
+                filterable={consumers && consumers.length > 5}
+                disabled={selectIsDisabled || !consumers || !consumers.length}
+                open={consumerSelectOpen}
+                onOpenChange={setConsumerSelectOpen}
+                // Although only one value could be selected
+                // Multiple type Select is used
+                // The reason - we need Select to be able to work with no value
+                // And it is easier to make multiple Select close on value change
+                // Than to enable single select to work with empty values
+                multiple
```

Comment:
For me it is not the best way to do it, it looks like a crutch. The better choice is to make a wrapper for `Select` that will allow to undo selection (the common way is - when selected item in Select popup is clicked again, selections cancels).

Reply:
Single select with ability to undo selection looks like a crutch as well, but it more difficult in terms of realisation. That's why I wrote this comment. Multiple Select crutch is much easier

Reply:
https://github.com/ydb-platform/ydb-embedded-ui/issues/386